apis/core/lib/domain/queries/table.ts

```diff
@@ -88,3 +88,16 @@ export async function getTableForColumnMapping(columnMapping: NamedNode, client
   }
   return results[0].table
 }
+
+export async function getCubeTable(csvMapping: CsvMapping): Promise<Term | undefined> {
+  const bindings = await SELECT`?table`
+    .WHERE`
+      GRAPH ?table {
+        ?table a ${cc.ObservationTable} .
+        ?table ${cc.csvMapping} ${csvMapping.id} .
+      }
+    `
+    .execute(parsingClient.query)
+
+  return bindings.shift()?.table
+}
```

Comment:
Because there is no way to access the other tables directly when create a new one, the domain code calls a SPARQL query to ensure a cube table does not already exist

Reply:
How are cube-tables identified? Is the `csvMapping.id`  discriminative attribute?

Reply:
I don't understand. `csvMapping.id` is that resource's URI. The data is that all tables are linked to their CSV Mapping resource. And only one of those tables can have the type `cc:ObservationTable` (notice old terminology, as now refer to them as cube tables)

Reply:
yes, I'm confused 

so, one can have:
- Many tables, each pointing to a csvMapping
- One of those tables is an Observation table

Is this always?

So this method wants to check for ObservationTables?

It could be named 

`currentObservationTable``